src/test/config/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+include_directories(${GLIB_INCLUDES})
```

Comment:
Optional, because I don't want to change the requirements on you again: :)

This sequence of tests validates that the config round-trips to *something* that shadow will parse, but wouldn't necessarily catch a lot of issues like dropping or changing parameters that the simple test here doesn't depend on.

It'd be nice to have tests that do a conversion and compare against an expected output. I see you have some extra yml and xml files still in here, that maybe are intended for such a purpose? If so, you could write such tests using `diff`, which returns a non-zero exit-code iff there is a diff. e.g.

```
COMMAND ${CMAKE_SOURCE_DIR}/src/tools/convert.py xml2yaml --output converted.yaml path/to/test-input.xml && diff path/to/test-expected-output.yaml converted.yaml)
```

Reply:
Yes ! I totally agree with you, I think it's the best way.
Unfortunately we can't run CMake test with multiple tests easily. So I used the XML and YAML files created previously in the test